spec/consumer_spec.rb

```diff
@@ -129,5 +131,21 @@
 
       expect(fetch_operation).to have_received(:fetch_from_partition).with("greetings", 42, anything)
     end
+
+    it "automatically resumes partitions if a timeout is set" do
+      time = Time.now
+
+      Timecop.freeze time do
+        consumer.pause("greetings", 0, timeout: 30)
+      end
+
+      Timecop.freeze time + 29 do
+        expect(consumer.paused?("greetings", 0)).to eq true
+      end
+
+      Timecop.freeze time + 31 do
+        expect(consumer.paused?("greetings", 0)).to eq false
+      end
+    end
```

Comment:
When using `travel`, I think time is still moving, so it's more difficult to make assumptions.